src/vehicle.h

```diff
@@ -843,6 +843,35 @@ class vehicle
         vehicle &operator=( const vehicle & ) = default;
 
     public:
+        enum class install_code : int {
+            SUCCESS,
+            // part doesn't exist
+            ERR_INVALID,
```

Comment:
these ERR_ prefixes are duplicating the functionality of ret_val's success bool

also ret_val&lt;void&gt; will do the same thing ret_val&lt;install_code&gt; does but with less code and no extra enum, unless you have a plan for more functionality on top of the enum

Reply:
Thank you greatly, these were all very good suggestions. I tried to enter you as a co-author on https://github.com/CleverRaven/Cataclysm-DDA/pull/65341/commits/28aa5852e863f7929745bf482372ac05fbb84954 after batching them (I thought the colorization would be more work than it was), but github didn't like my formatting. I hope that's fine?

Reply:
Don't worry about it, just a thanks is enough